### PR TITLE
Stop searching platform groups after first match.

### PR DESCRIPTION
**Pull request by wxtim**
Platform selection from a platform group was missing a `break` in the `for` loop, leading to the possibility of the wrong platform being selected.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by MetRonnie** [comment]
Kicking tests

File: tests/unit/test_platforms_get_platform.py [hunk 1]
```diff
@@ -237,18 +237,25 @@ def test_get_platform_using_platform_name_from_job_info(
...
-                hosts = bet
+            [[aleph, bet, alpha, beta]]
 
         [platform groups]
             [[hebrew_letters]]
+                platforms = alpha, beta
+                [[[selection]]]
+                    method = definition order
+            [[aleph]]
+                platforms = alpha
```
Comment (MetRonnie) [inline tests/unit/test_platforms_get_platform.py:257]: This `aleph` platform group is a little confusing to me

**Comment by MetRonnie** [comment]
Ah, wrong base branch. Need to rebase onto 8.1.x

File: tests/unit/test_platforms_get_platform.py [hunk 2]
```diff
@@ -237,18 +237,25 @@ def test_get_platform_using_platform_name_from_job_info(
...
-            [[bet]]
-                hosts = bet
+            [[aleph, bet, alpha, beta]]
 
         [platform groups]
             [[hebrew_letters]]
+                platforms = alpha, beta
+                [[[selection]]]
+                    method = definition order
+            [[aleph]]
```
Comment (MetRonnie) [inline tests/unit/test_platforms_get_platform.py:256]:
```suggestion
            [[aleph]]
            # Group with same name as platform to try and
            # trip up the platform selection logic after it
            # has processed [[.*_letters]] below
```

**Review by MetRonnie — COMMENTED** [review]
I have suggested a comment in light of your explanation, to clarify the test

Can you add a description of the bug to the PR body?

File: tests/unit/test_platforms_get_platform.py (hunk 2 above)
Comment (wxtim) [reply tests/unit/test_platforms_get_platform.py:256]: Yes. That's a good idea, this test is wierd and non obvious
```suggestion
            # Group with same name as platform to try and
            # trip up the platform selection logic after it
            # has processed [[.*_letters]] below
            # https://github.com/cylc/cylc-flow/pull/5398#discussion_r1130692633
```

File: tests/unit/test_platforms_get_platform.py [hunk 3]
```diff
@@ -237,18 +237,28 @@ def test_get_platform_using_platform_name_from_job_info(
...
+            # Group with same name as platform to try and
+            # trip up the platform selection logic after it
+            # has processed [[.*_letters]] below
```
Comment (wxtim) [inline tests/unit/test_platforms_get_platform.py:259]: Yes. That's a good idea, this test is wierd and non obvious
Feel free to add the following if you like.
```suggestion
            [[aleph]]
            # Group with same name as platform to try and
            # trip up the platform selection logic after it
            # has processed [[.*_letters]] below
            # https://github.com/cylc/cylc-flow/pull/5398#discussion_r1130692633
```

**Review by MetRonnie — APPROVED** [review]

**Review by hjoliver — APPROVED** [review]
👍

**Pull request merged**